tests/components/upnp/test_config_flow.py

```diff
@@ -137,23 +302,51 @@ async def test_flow_import(hass: HomeAssistant):
     assert result["data"] == {
         CONFIG_ENTRY_ST: TEST_ST,
         CONFIG_ENTRY_UDN: TEST_UDN,
-        CONFIG_ENTRY_HOSTNAME: TEST_HOSTNAME,
+        CONFIG_ENTRY_ORIGINAL_UDN: TEST_UDN,
+        CONFIG_ENTRY_LOCATION: TEST_LOCATION,
+        CONFIG_ENTRY_MAC_ADDRESS: TEST_MAC_ADDRESS,
     }
 
 
+@pytest.mark.usefixtures("mock_get_source_ip")
+async def test_flow_import_incomplete_discovery(hass: HomeAssistant):
```

Comment:
Looks like this test is doing I/O
`4.01s call     tests/components/upnp/test_config_flow.py::test_flow_import_incomplete_discovery[pyloop]`
`pytest --durations=10 --cov=homeassistant/components/upnp/ --cov-report term-missing -- tests/components/upnp/`

Reply:
Coverage is almost there
homeassistant/components/upnp/config_flow.py       115      4    97%   57, 138, 246, 250

Reply:
> Well... I thought we were there but it looks like we need a bit more coverage to merge this.
> 
> `homeassistant/components/upnp/config_flow.py 115 9 92% 57, 138, 162, 206-207, 246, 250, 265-266`

Line 138 can never be reached. The `user_input` is guarded by a voluptuous schema, accepting only a `unique_id` for a discovery. Is replacing the check + return with an assert ok?

Reply:
Meh, didn't mean to reply here...